core/src/main/scala/magnolia1/magnolia.scala

```diff
@@ -944,11 +944,46 @@ private[magnolia1] object CompileTimeState {
   }
 }
 
-object CallByNeed { def apply[A](a: => A): CallByNeed[A] = new CallByNeed(() => a) }
-final class CallByNeed[+A](private[this] var eval: () => A) extends Serializable {
+object CallByNeed {
+
+  /** Initializes a class that allows for suspending evaluation of a value until it is needed. Evaluation of a value via `.value` can only
```

Comment:
Semantics here don't change, but perhaps the comment can be worded in a way that's clearer? Previous functionality also has evaluation/computation only occurring once, but of course it can be called multiple times.